### DISCUSSION TITLE
Vs Code Go extension deleting my code. . .  Help! 😥

### DISCUSSION THREAD

**Discussion by Major2000** [golang/vscode-go#2490] [Help] [answered]
I installed the official go extension in VS Code all the necessary executables are installed, the problem is when I am writing my code their is this weird thing that deletes my code all the time and  I cannot stop it....

 

https://user-images.githubusercontent.com/65360332/193589270-3600f85d-528e-4770-b864-fec1c5503e9d.mp4

I think its something to do with the settings but I don't know how to set them.... 🥵

**Comment by suzmue** [comment, marked as answer]
Deleting unused imports is [the default behavior](https://github.com/golang/vscode-go/blob/master/docs/features.md#format-and-organize-imports) to help saved files can compile - In Go, unused imports are compile-time errors.

For different behavior, follow the instruction in ["Formatting Code and Organizing Imports"](https://github.com/golang/vscode-go/blob/master/docs/advanced.md#formatting-code-and-organizing-imports).

"[go]": { "editor.codeActionsOnSave": { "source.organizeImports": false }}

**Reply by d9j** [reply]
why it's enabled by default., i really dont get it. like it's not a bug it is a feature. however it only golang extension enforces it

**Reply by ingframin** [reply]
Wouldn't it be better to simply comment out the unused imports rather than deleting them?
